generic-pretty-instances/test/GenericPrettyInstancesSpec.hs

```diff
@@ -1,12 +1,30 @@
+{-# LANGUAGE StandaloneDeriving #-}
+
 module GenericPrettyInstancesSpec (spec) where
 
 import Test.Hspec
+import Text.PrettyPrint.GenericPretty (Out (..))
 import Text.PrettyPrint.GenericPretty.Util
   ( inspect,
     inspectPlain,
   )
 import Universum hiding (show)
 
+newtype Username = Username String
+
+deriving stock instance Generic Username
+
+instance Out Username
+
+data Configuration = Configuration
+  { username :: Username,
+    localHost :: String
+  }
```

Comment:
For both data and newtype you can derive Generic just in place where it's defined:
```haskell
data NodeUri = NodeUri
  { nodeUriPubKey :: NodePubKey,
    nodeUriSocketAddress :: SocketAddress
  }
  deriving stock
    ( Eq,
      Ord,
      Show,
      Generic
    )
```